### PR TITLE
Promote catalog-filesystem out of incubator

### LINKED ISSUE #1068: Promote `filesystem` catalog out of incubator
**Describe the solution you'd like**
Promote the `filesystem` catalog module out of the incubator project

### PR DESCRIPTION
**Pull request by attilakreiner**
## Description

This change promotes the filesystem catalog module out of the incubator project.

Fixes #1068

### CODE REVIEW

File: runtime/catalog-filesystem/src/main/java/io/aklivity/zilla/runtime/catalog/filesystem/internal/FilesystemCatalogFactorySpi.java
Comment (jfallows) [inline runtime/catalog-filesystem/src/main/java/io/aklivity/zilla/runtime/catalog/filesystem/internal/FilesystemCatalogFactorySpi.java:1]: Need to remove `@Incubating` annotation.

File: runtime/pom.xml
Comment (jfallows) [inline runtime/pom.xml:1]: Need to add
```
    <module>catalog-filesystem</module>
```

**Review by jfallows — CHANGES_REQUESTED** [review]

File: runtime/catalog-filesystem/src/main/java/io/aklivity/zilla/runtime/catalog/filesystem/internal/FilesystemCatalogFactorySpi.java
Comment (attilakreiner) [reply runtime/catalog-filesystem/src/main/java/io/aklivity/zilla/runtime/catalog/filesystem/internal/FilesystemCatalogFactorySpi.java:1]: done

File: runtime/pom.xml
Comment (attilakreiner) [reply runtime/pom.xml:1]: done

**Review by jfallows — APPROVED** [review]

**Pull request merged**